src/main/java/com/datadoghq/sketch/ddsketch/mapping/QuarticallyInterpolatedMapping.java

```diff
@@ -0,0 +1,76 @@
+/* Unless explicitly stated otherwise all files in this repository are licensed under the Apache License 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2021 Datadog, Inc.
+ */
+
+package com.datadoghq.sketch.ddsketch.mapping;
+
+/**
+ * A fast {@link IndexMapping} that approximates the memory-optimal one (namely {@link
+ * LogarithmicMapping}) by extracting the floor value of the logarithm to the base 2 from the binary
+ * representations of floating-point values and quartically interpolating the logarithm in-between.
+ *
+ * <p>The optimal polynomial coefficients can be calculated using the method described in {@link
+ * CubicallyInterpolatedMapping}.
+ */
+public class QuarticallyInterpolatedMapping extends LogLikeIndexMapping {
+
+  private static final double A = -2.0 / 25.0;
+  private static final double B = 8.0 / 25.0;
+  private static final double C = -17.0 / 25.0;
+  private static final double D = 36.0 / 25.0;
+
+  public QuarticallyInterpolatedMapping(double relativeAccuracy) {
+    super(relativeAccuracy);
+  }
+
+  /** {@inheritDoc} */
+  QuarticallyInterpolatedMapping(double gamma, double indexOffset) {
+    super(gamma, indexOffset);
+  }
+
+  @Override
+  double log(double value) {
+    final long longBits = Double.doubleToRawLongBits(value);
+    final double s = DoubleBitOperationHelper.getSignificandPlusOne(longBits) - 1;
+    final double e = (double) DoubleBitOperationHelper.getExponent(longBits);
+    return (((A * s + B) * s + C) * s + D) * s + e;
+  }
+
+  @Override
+  double logInverse(double index) {
+    final double exponent = Math.floor(index);
+    final double e = exponent - index;
+    // Derived from Ferrari's method
+    final double alpha = -(3 * B * B) / (8 * A * A) + C / A; // 2.5
+    final double beta = (B * B * B) / (8 * A * A * A) - (B * C) / (2 * A * A) + D / A; // -9.0
+    final double gamma =
+        -(3 * B * B * B * B) / (256 * A * A * A * A)
+            + (C * B * B) / (16 * A * A * A)
+            - (B * D) / (4 * A * A)
+            + e / A;
+    final double p = -(alpha * alpha) / 12 - gamma;
+    final double q = -(alpha * alpha * alpha) / 108 + (alpha * gamma) / 3 - (beta * beta) / 8;
+    final double r = -q / 2 + Math.sqrt((q * q) / 4 + (p * p * p) / 27);
```

Comment:
I'm curious if you can do some manual common subexpression elimination here and get a slight improvement.

Reply:
It probably all folds away, but still curious

Reply:
I had a look at the bytecode and it seems to fold away, which is why I kept explicit expressions, most notably for `alpha` and `beta`. Beyond what's achieved with that folding, I don't see an obvious way to rewrite these expressions in a less computationally expensive way (although I may have missed it). Furthermore, one advantage of the current form is that it exactly matches [Wikipedia's solution](https://en.wikipedia.org/wiki/Quartic_equation#Summary_of_Ferrari's_method), which makes validating correctness easier.

```
  double logInverse(double);
    Code:
       0: dload_1
       1: invokestatic  #15                 // Method java/lang/Math.floor:(D)D
       4: dstore_3
       5: dload_3
       6: dload_1
       7: dsub
       8: dstore        5
      10: ldc2_w        #16                 // double 2.500000000000001d
      13: dstore        7
      15: ldc2_w        #18                 // double -9.0d
      18: dstore        9
      20: ldc2_w        #20                 // double -12.5d
      23: dload         5
      25: ldc2_w        #7                  // double -0.08d
      28: ddiv
      29: dadd
      30: dstore        11
      32: ldc2_w        #22                 // double -0.5208333333333337d
      35: dload         11
      37: dsub
      38: dstore        13
      40: ldc2_w        #24                 // double -0.14467592592592607d
      43: ldc2_w        #16                 // double 2.500000000000001d
      46: dload         11
      48: dmul
      49: ldc2_w        #26                 // double 3.0d
      52: ddiv
      53: dadd
      54: ldc2_w        #28                 // double 10.125d
      57: dsub
      58: dstore        15
      60: dload         15
      62: dneg
      63: ldc2_w        #30                 // double 2.0d
      66: ddiv
      67: dload         15
      69: dload         15
      71: dmul
      72: ldc2_w        #32                 // double 4.0d
      75: ddiv
      76: dload         13
      78: dload         13
      80: dmul
      81: dload         13
      83: dmul
      84: ldc2_w        #34                 // double 27.0d
      87: ddiv
      88: dadd
      89: invokestatic  #36                 // Method java/lang/Math.sqrt:(D)D
      92: dadd
      93: dstore        17
      95: dload         17
      97: invokestatic  #37                 // Method java/lang/Math.cbrt:(D)D
     100: dstore        19
     102: ldc2_w        #38                 // double -2.083333333333334d
     105: dload         19
     107: dadd
     108: dload         13
     110: ldc2_w        #26                 // double 3.0d
     113: dload         19
     115: dmul
     116: ddiv
     117: dsub
     118: dstore        21
     120: ldc2_w        #16                 // double 2.500000000000001d
     123: ldc2_w        #30                 // double 2.0d
     126: dload         21
     128: dmul
     129: dadd
     130: invokestatic  #36                 // Method java/lang/Math.sqrt:(D)D
     133: dstore        23
     135: dconst_1
     136: dload         23
     138: ldc2_w        #40                 // double 7.500000000000003d
     141: ldc2_w        #30                 // double 2.0d
     144: dload         21
     146: dmul
     147: dadd
     148: ldc2_w        #42                 // double -18.0d
     151: dload         23
     153: ddiv
     154: dadd
     155: dneg
     156: invokestatic  #36                 // Method java/lang/Math.sqrt:(D)D
     159: dsub
     160: ldc2_w        #30                 // double 2.0d
     163: ddiv
     164: dadd
     165: dstore        25
     167: dload_3
     168: d2l
     169: dload         25
     171: dconst_1
     172: dadd
     173: invokestatic  #44                 // Method com/datadoghq/sketch/ddsketch/mapping/DoubleBitOperationHelper.buildDouble:(JD)D
     176: dreturn
```